Lib/functools.py

```diff
@@ -949,6 +952,11 @@ def _method(*args, **kwargs):
         _method.__isabstractmethod__ = self.__isabstractmethod__
         _method.register = self.register
         update_wrapper(_method, self.func)
+
+        # not all objects have __dict__ (e.g. classes with __slots__)
+        if hasattr(obj, '__dict__'):
+            obj.__dict__[self._attrname] = _method
```

Comment:
https://github.com/python/cpython/pull/107148#discussion_r1272094789

Please leave a comment that self._attrname is always set at this moment.

Reply:
Would be great if you could also add a comment about why we're inserting it into `__dict__` at all! (Performance!)